src/core/expr/fexpr_round.cc

```diff
@@ -0,0 +1,384 @@
+                          std::move(col), stype
+                      ));
+      }
+      if (ndigits_ > 0) {
+        return Column(new RoundPos_ColumnImpl<double>(
+                            std::move(col),
+                            std::pow(10.0, ndigits_)
+                      ));
+      }
+      else {
+        return Column(new RoundNeg_ColumnImpl<double>(
+                            std::move(col),
+                            std::pow(10.0, -ndigits_)
+                      ));
+      }
+    }
+};
+
+
+
+
+//------------------------------------------------------------------------------
+// Python-facing `round()` function
+//------------------------------------------------------------------------------
+
+static const char* doc_round =
+R"(round(cols, ndigits=None)
+--
+
+Round the values in `cols` up to the specified number of the digits
+of precision `ndigits`. If the number of digits is omitted, rounds
+to the nearest integer.
+
+The parameter `ndigits` can be either
+
+Generally, this operation is equivalent to::
+
+    rint(col * 10**ndigits) / 10**ndigits
+
+where function `rint()` rounds to the nearest integer.
+
+Parameters
+----------
+cols: FExpr
+    Input data for rounding. This could be an expression yielding
+    either a single or multiple columns. The `round()` function will
+    apply to each column independently and produce as many columns
+    in the output as there were in the input.
+
+    Only numeric columns are allowed: boolean, integer or floating.
+    An exception will be raised if `cols` contains a non-numeric
+    column.
+
+ndigits: int | None
+    The number of precision digits to retain. This parameter could
+    be either positive or negative (or None). If positive then it
+    gives the number of digits after the decimal point. If negative,
+    then it rounds the result up to the corresponding power of 10.
+
+    For example, `123.45` rounded to `ndigits=1` is `123.4`, whereas
+    rounded to `ndigits=-1` it becomes `120.0`.
+
+return: FExpr
+    f-expression that rounds the values in its first argument to
+    the specified number of precision digits.
+
+    Each input column will produce the column of the same stype in
+    the output; except for the case when `ndigits` is `None` and
+    the input is either `float32` or `float64`, in which case an
+    `int64` column is produced (similarly to python's `round()`).
+
+Notes
+-----
+Values that are exactly half way in between their rounded neighbors
+are converted towards their nearest even value. For example, both
+`7.5` and `8.5` are rounded into `8`, whereas `6.5` is rounded as `6`.
+
+Rounding integer columns may produce unexpected results for values
+that are close to the min/max value of that column's storage type.
+For example, when an `int8` value `127` is rounded to nearest 10, it
+becomes `130`. However, since `130` cannot be represented as `int8` a
+warp-around occurs and the result becomes `-126`.
```

Comment:
wrap-around?

Also, what about promoting column to `int16` in this case?

Reply:
Then the stype of the result would not be the function of the stype of the input anymore, and it opens a HUGE can of worms. On the other hand, numpy also does the same thing as described here:
```
>>> np.array([127], dtype='int8').round(decimals=-1)
array([-126], dtype=int8)
```

Reply:
If input is `int8` then we can always promote to `int16`, etc, unrelated to the actual data. Otherwise it seems very dangerous to me. No one really cares about the stypes, but when `round(127) == -126` this could make a difference.

Reply:
It is reasonable to promote int8 or int16 into int32, but it will fix the problem only for those stypes. Obviously, int32 values near the INT32_MAX will have the same issues. And int64 values near the INT64_MAX.

Reply:
`int32` can be promoted to `int64`, but for `int64` yes, there is no way... but this is a hard limit

Reply:
I don't think the promotion `int32` -> `int64` is desirable. After all, most users store integers that are of reasonable magnitude, far away from the boundaries. Doubling the column's storage requirements to potentially handle few corner cases is just not worth it.

Reply:
>most users store integers that are of reasonable magnitude

I guess this is true for `int32` and `int64`, but may not be true for `int8` and `int16`. For instance, `int8` can be automatically assigned by `fread` during data import.

>Doubling the column's storage requirements to potentially handle few corner cases is just not worth it.

Well, this column is virtual anyways.

Reply:
I don't think fread creates int8 columns right now; however we may do so in the future. But even if we do, it will only be for a 0/1 column only